mithril-client/src/dependencies/builder.rs

```diff
@@ -181,10 +182,18 @@ impl DependenciesBuilder {
 
     async fn build_snapshot_service(&mut self) -> StdResult<Arc<dyn SnapshotService>> {
         let snapshot_service = MithrilClientSnapshotService::new(
-            self.get_snapshot_client().await?,
-            self.get_certificate_client().await?,
-            self.get_certificate_verifier().await?,
-            self.get_immutable_digester().await?,
+            self.get_snapshot_client()
+                .await
+                .with_context(|| "Dependencies Builder can not get Snapshot Client")?,
```

Comment:
Shouldn't those contextual error message take place in the called methods `get_something` in the builder?